src/Functions/ReplaceRegexpImpl.h

```diff
@@ -23,7 +27,19 @@ struct ReplaceRegexpTraits
     };
 };
 
-/** Replace all matches of regexp 'needle' to string 'replacement'. 'needle' and 'replacement' are constants.
+static constexpr ReplaceStringTraits::Replace convertToReplaceStringTraits(ReplaceRegexpTraits::Replace replace)
+{
+    switch (replace)
+    {
+        case ReplaceRegexpTraits::Replace::First:
+            return ReplaceStringTraits::Replace::First;
+        case ReplaceRegexpTraits::Replace::All:
+            return ReplaceStringTraits::Replace::All;
+    }
+    UNREACHABLE();
```

Comment:
`std::unreachable`. But you would not even need that if you use a `switch` here.

Reply:
It is needed in case new enum item is added under `ReplaceRegexpTraits::Replace` in future.